packages/wp-story-editor/src/components/postLock/postLock.js

```diff
@@ -124,8 +122,7 @@ function PostLock() {
 
   useEffect(() => {
     if (enablePostLocking && showLockedDialog && currentUserLoaded) {
-      if (lockUser?.id !== currentUser.id) {
```

Comment:
> Why does this fix the problem? What was the original problem?

@spacedmonkey This part was the underlying problem.

`lockUser` is always an object, with `lockUser.id` being `0` if there is no user.

So `lockUser?.id !== currentUser.id` is always `true`

Reply:
> lockUser is always an object

This is a bit confusing. Is making `lockUser` nullable possible?

> lockUser.id being 0 if there is no user.

Could we change the fallback `id` to `NaN`?